src/node/rpc/memberfrontend.h

```diff
@@ -639,25 +662,29 @@ namespace ccf
         auto verifier = tls::make_verifier(
           std::vector<uint8_t>(args.rpc_ctx->session.caller_cert));
         const auto rs = args.rpc_ctx->get_params().get<RawSignature>();
-        if (!verifier->verify(last_ma->next_nonce, rs.sig))
+
+        if (!verifier->verify(
+              ma->next_state_digest.h.data(),
+              ma->next_state_digest.h.size(),
+              rs.sig.data(),
+              rs.sig.size()))
         {
           args.rpc_ctx->set_response_error(
             jsonrpc::StandardErrorCodes::INVALID_PARAMS,
             "Signature is not valid");
           return;
         }
 
-        MemberAck next_ma{rs.sig, rng->random(SIZE_NONCE)};
-        mas->put(args.caller_id, next_ma);
+        MemberAck next_ma{rs.sig, sig_view->get(0)->root};
```

Comment:
Similarly to above - should we be updating the ack digest here? It seems we _could_ be storing these values in separate tables (one is updated by `updateAckStateDigest`, the other by `ack` - they don't need to be in the same struct), but if we're storing them side-by-side then they should at least be a matching pair (ie - a sig, and the root it signs over).

Reply:
As discussed, we now store the full signed request, containing the digest of the state along with the next state digest to-be-signed.